ete4/gtdb_taxonomy/gtdbquery.py

```diff
@@ -565,18 +566,24 @@ def annotate_tree(self, t, taxid_attr="name", tax2name=None, tax2track=None, tax
         for n in t.traverse('postorder'):
             try:
                 #node_taxid = int(getattr(n, taxid_attr))
-                node_taxid = getattr(n, taxid_attr)
+                node_taxid = n.props.get(taxid_attr)
             except (ValueError, AttributeError):
                 node_taxid = None
 
             n.add_prop('taxid', node_taxid)
+            
             if node_taxid:
                 tmp_taxid = self.get_name_translator([node_taxid]).get(node_taxid, [None])[0]
                 #tmp_taxid = self.get_name_translator([node_taxid])[node_taxid][0] # translate to temperatoru
                 if node_taxid in merged_conversion:
                     node_taxid = merged_conversion[node_taxid]
 
-                n.add_props(sci_name = tax2name.get(node_taxid, getattr(n, taxid_attr, '')),
+                rank = tax2rank.get(tmp_taxid, 'Unknown')
+                if rank =='subspecies':
+                    sci_name = tax2name.get(tax2track.get(tmp_taxid,[])[-2], '')
+                else:
+                    sci_name = tax2name.get(node_taxid, getattr(n, taxid_attr, ''))
```

Comment:
Ideally I'd like to see a comment explaining why the sci_name has to be found in this way.